src/protocol/utils.rs

```diff
@@ -1,9 +1,44 @@
 use std::{net::SocketAddr, time::SystemTime};
+use std::net::IpAddr;
 use aquatic_udp_protocol::ConnectionId;
 
+// todo: SALT should be randomly generated on startup
+const SALT: &str = "SALT";
```

Comment:
hi @WarmBeer should we do it in the constructor (TorrentTracker::new)? or maybe in the main:

```rust
// Initialize Torrust tracker
let salt = "SALT";
let tracker = match TorrentTracker::new(config.clone()) {
    Ok(tracker) => Arc::new(tracker, salt),
    Err(error) => {
        panic!("{}", error)
    }
};
```

I would also inject it `get_connection_id` so we can test it.

Reply:
Hey @josecelano ,

I think the generation of the salt could be moved here:

https://github.com/torrust/torrust-tracker/blob/main/src/udp/server.rs#L27
```rust
pub async fn start(&self) {
        loop {
            let mut data = [0; MAX_PACKET_SIZE];
            let socket = self.socket.clone();
            let tracker = self.tracker.clone();

            tokio::select! {
                _ = tokio::signal::ctrl_c() => {
                    info!("Stopping UDP server: {}..", socket.local_addr().unwrap());
                    break;
                }
                Ok((valid_bytes, remote_addr)) = socket.recv_from(&mut data) => {
                    let payload = data[..valid_bytes].to_vec();

                    debug!("Received {} bytes from {}", payload.len(), remote_addr);
                    debug!("{:?}", payload);

                    let response = handle_packet(remote_addr, payload, tracker).await;
                    UdpServer::send_response(socket, remote_addr, response).await;
                }
            }
        }
    }
```

And since all connection id functions are only relevant for UDP, they can also be moved to the UDP module like you already said.